RPI/metabot.py

```diff
@@ -0,0 +1,270 @@
+            self.lasty = y
+            self.lastdmh = dmh
+            return {"x": x, "y": y, "dmh" : dmh}
+        else :
+            return {}
+
+    @staticmethod
+    def deadzone(val, cutoff):
+        if abs(val) < cutoff:
+            return 0
+        elif val > 0:
+            return val - cutoff
+        else:
+            return val + cutoff
+
+    def run(self):
+        logger.info("Controller Thread Running")
+        port = 10000
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(1.0)
+        s.bind((self._ipaddress, port))
+        addr = None
+
+        logger.info("Waiting for Controller Connection")
+        while self.running:
+            s.listen(1)
+            try:
+                self.conn, addr = s.accept()
+                self.conn.setblocking(1)
+            except socket.timeout:
+                pass
+            if self.conn is not None:
+                logger.info("Controller Connected to {0}".format(addr))
+            while self.running and self.conn is not None:
+                joystick_msg = self.joystick_update()
+                self._controlLoop.controller_update(joystick_msg)
+
+        logger.info("Controller Thread Stopped")
+
+
+class Arduino(pykka.ThreadingActor):
+    """ Represents the connection to the Arduino over a serial connection """
+
+    def __init__(self):
+        logger.info("Arduino Actor Running")
+        super(Arduino, self).__init__()
+        self.ser = None
+        self.open_serial()
+
+    def open_serial(self):
+        try:
+            self.ser = serial.Serial(port="/dev/ttyAMA0",
+                                     baudrate=115200,
+                                     parity=serial.PARITY_NONE,
+                                     stopbits=serial.STOPBITS_ONE,
+                                     bytesize=serial.EIGHTBITS)
+            self.ser.close()
+            self.ser.open()
+        except serial.SerialException:
+            logger.warning("Failed to Open Serial Connection")
+            self.ser = None
+
+    def on_stop(self):
+        logger.info("Arduino Actor Stopped")
+
+    def send_cmd(self, cmd):
+        logger.info("sending: {}".format(cmd))
+        if self.ser is not None:
+            try:
+                self.ser.write(cmd + "\r")
+            except serial.SerialException:
+                logger.warning("Serial Write Failed")
+                self.ser.close()
+                self.ser = None
+        else:
+            self.open_serial()
+
+
+class ControlLoop(pykka.ThreadingActor):
+    """ Control loop that reads inputs and drives outputs """
+
+    def __init__(self, arduino):
+        logger.info("Control Loop Actor Running")
+        super(ControlLoop, self).__init__()
+        self._arduino = arduino
+
+    @staticmethod
+    def _xytolr(x, y):
+        speed = y * 1000.0
+        direction = x * 200.0
+        l = int(speed + direction)
+        r = int(speed - direction)
+        return l, r
+
```

Comment:
Does the _xytolr() function do what the old def _update_motor_speeds() function did?  If not, its going to change the control 'feel' of the robot.  

I think I want to try this out before we decide which algorithm to use here